odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,165 @@
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Implementation of the Forward-Backward splitting algorithm """
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator import Operator
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """ The forward-backward primal-dual splitting algorithm
+
+    Minimizes the sum of several convex functions composed with linear
+    operators
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``
+
+    Where f, g_i, h are convex functions, L_i are linear `Operator`'s, and h is
+    also differentiable.
+
+    Can also be used to solve the more general problem
+
+        ``min_x f(x) + sum_i (g_i @ l_i)(L_i x)``
+
+    where l_i are convex functions and @ is the infimal convolution:
+
+        ``(f @ g)(x) = inf_y { f(x-y) + g(y) }``
+
+    Parameters
+    ----------
+    x : `Vector`
+        Initial point
+    prox_f : `callable`
+        Funciton returning an `Operator` when called with `tau`.
+        The Operator should be the proximal of `tau * f`.
+    prox_cc_g : `sequence` of `callable`'s
+        Sequence of functions returning an operator when called with stepsize
+        `sigma`.
+        The Operator should be the proximal of `sigma_i * g_i^*`.
+    grad_h : `Operator`
+        Operators representing the gradient of  `h`.
+    L : `sequence` of `Operator`'s
+        A sequence with as many elements as ``prox__cc_gs`` of operators
+        ``L_i``
+    tau : `float`
+        Stepsize of ``f``
+    sigma : `sequence` of  `float`
+        Stepsize of the ``g_i``'s
+    niter : `int`
+        Number of iterations
+    callback : `Callback`, optional
+        Show partial results
+
+    Other Parameters
+    ----------------
+    grad_cc_l : `sequence` of `Operator`'s, optional
+        Sequence of operators representing the gradient of  `l_i^*`.
+        If omitted, the simpler problem will be considered, which corresponds
+        to the convex conjugate functionals of the l_i:s being zero functionals
+        and hence the gradient being the zero-operator.
+
+    Notes
+    -----
+    Strictly, we have the following conditions on the functions involved: f and
+    g_i are proper, convex and lower semicontinuous, and h is convex and
+    differentialbe with :math:`\\eta`-Lipschitz continuous gradient.
+
+    For the optional input l_i we need it to be proper, convex, lower
```

Comment:
The "we" style is not good for documentation (one can debate if it's good for papers or not). It can always be formulated more elegantly. My suggestion is 
```
The optional functionals :math:`l_i` need to be ...
```